src/gui/sensor/qgssensorguiregistry.h

```diff
@@ -0,0 +1,269 @@
+
+    QIcon mIcon;
+    QgsSensorWidgetFunc mWidgetFunc = nullptr;
+    QgsSensorCreateFunc mCreateFunc = nullptr;
+
+};
+
+#endif
+
+/**
+ * \ingroup gui
+ * \class QgsSensorGuiRegistry
+ * \brief Registry of available sensor GUI behavior.
+ *
+ * QgsSensorGuiRegistry is not usually directly created, but rather accessed through
+ * QgsGui::sensorGuiRegistry().
+ *
+ * This acts as a companion to QgsSensorRegistry, handling only
+ * the components related to the GUI behavior of sensors.
+ *
+ * \since QGIS 3.32
+ */
+class GUI_EXPORT QgsSensorGuiRegistry : public QObject
+{
+    Q_OBJECT
+
+  public:
+
+    /**
+     * Creates a new empty sensor GUI registry.
+     *
+     * QgsSensorGuiRegistry is not usually directly created, but rather accessed through
+     * QgsGui::sensorGuiRegistry().
+    */
+    QgsSensorGuiRegistry( QObject *parent = nullptr );
+    ~QgsSensorGuiRegistry() override;
+
+    //! QgsSensorGuiRegistry cannot be copied.
+    QgsSensorGuiRegistry( const QgsSensorGuiRegistry &rh ) = delete;
+    //! QgsSensorGuiRegistry cannot be copied.
+    QgsSensorGuiRegistry &operator=( const QgsSensorGuiRegistry &rh ) = delete;
+
+    /**
+     * Populates the registry with standard sensor types. If called on a non-empty registry
+     * then this will have no effect and will return FALSE.
+     */
+    bool populate();
+
+    /**
+     * Returns the metadata for the specified sensor \a type. Returns NULLPTR if
+     * a corresponding sensor type was not found in the registry.
+     */
+    QgsSensorAbstractGuiMetadata *sensorMetadata( const QString &type ) const;
+
+    /**
+     * Registers the GUI metadata for a new sensor type. Takes ownership of the metadata instance.
+     */
+    bool addSensorGuiMetadata( QgsSensorAbstractGuiMetadata *metadata SIP_TRANSFER );
+
+    /*
+     * IMPORTANT: While it seems like /Factory/ would be the correct annotations here, that's not
+     * the case.
+     * As per Phil Thomson's advice on https://www.riverbankcomputing.com/pipermail/pyqt/2017-July/039450.html:
+     *
+     * "
+     * /Factory/ is used when the instance returned is guaranteed to be new to Python.
+     * In this case it isn't because it has already been seen when being returned by QgsProcessingAlgorithm::createInstance()
+     * (However for a different sub-class implemented in C++ then it would be the first time it was seen
+     * by Python so the /Factory/ on create() would be correct.)
+     *
+     * You might try using /TransferBack/ on create() instead - that might be the best compromise.
+     * "
+     */
+
+    /**
+     * Creates a new instance of a sensor given the \a type.
+     */
+    QgsAbstractSensor *createSensor( const QString &type, QObject *parent = nullptr ) const SIP_TRANSFERBACK;
+
+    /**
+     * Creates a new instance of a sensor configuration widget for the specified \a sensor.
+     */
+    QgsAbstractSensorWidget *createSensorWidget( QgsAbstractSensor *sensor ) const SIP_TRANSFERBACK;
```

Comment:
Can you add a warning that the sensor needs to live for the duration of the widget? (And is that criteria currently fulfilled?)